server/src/main/java/com/thoughtworks/go/config/update/PartialConfigUpdateCommand.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config.update;
+
+import com.rits.cloning.Cloner;
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.config.remote.PartialConfig;
+
+public class PartialConfigUpdateCommand implements UpdateConfigCommand {
+    private static final Cloner CLONER = new Cloner();
+
+    private final PartialConfig partial;
+    private final String fingerprint;
+    private final PartialConfigResolver resolver;
+
+    public PartialConfigUpdateCommand(final PartialConfig partial, final String fingerprint, PartialConfigResolver resolver) {
+        this.partial = partial;
+        this.fingerprint = fingerprint;
+        this.resolver = resolver;
+    }
+
+    @Override
+    public CruiseConfig update(CruiseConfig cruiseConfig) {
+        if (partial != null && fingerprint != null) {
+            cruiseConfig.getPartials().remove(resolver.findPartialByFingerprint(cruiseConfig, fingerprint));
```

Comment:
`findPartialByFingerprint` can be null. Is that going to be a problem? In some race condition maybe?

Reply:
Not sure - this was a refactor (move method, rename) so this was likely a problem before. I can add a null check just in case anyway.